packages/@aws-cdk/aws-apigatewayv2/lib/http/api-mapping.ts

```diff
@@ -61,18 +61,44 @@ export class HttpApiMapping extends Resource implements IApiMapping {
    */
   public readonly apiMappingId: string;
 
+  /**
+   * API Mapping key
+   */
+  public readonly mappingKey?: string;
+
   constructor(scope: Construct, id: string, props: HttpApiMappingProps) {
     super(scope, id);
 
+    if ((!props.stage?.stageName) && !props.api.defaultStage) {
+      throw new Error('stage is required if default stage is not available');
+    }
+
+    const paramRe = '^[a-zA-Z0-9]+[$-_.+!*\'\(\),]?$';
+    if (props.apiMappingKey && !new RegExp(paramRe).test(props.apiMappingKey)) {
+      throw new Error('An ApiMapping key may contain only letters, numbers and one of $-_.+!*\'(),');
+    }
```

Comment:
Can you add a unit test to make sure this regex behaves exactly as you expect?

Reply:
Thanks for the review.

I just added tests on `ApiMappingKey` against the following conditions:

'/'
'/foo'
'foo/bar'
'foo/'
'^foo'
'foo.*$'